### PR TITLE
Add IsArray and IsPureArray checking config is an array

### LINKED ISSUE #44: Query Config for being an array
Add support for asking Config to be an array.

### PR DESCRIPTION
**Pull request by urso**
Closes #44 

- requires #43 to adapt unit test

### CODE REVIEW

**Comment by ruflin** [comment]
LGTM. Not 100% what do you mean by `PureArray` ?

**Comment by urso** [comment]
in ucfg a `*Config` can be a map and an array at the same time to gain some flexibility. The IsArray method checks it's and array, but does not exclude the fact the config potentially being a map at the same time. The IsPureArray checks the Config being an array and have any named properties (is no dict/map/object).

**Comment by urso** [comment]
changed implementation to provide `IsArray` and `IsDict`. `IsPureArray` has been removed in favor of users writing `cfg.IsArray() && !cfg.IsDict()`.

File: types.go [hunk 1]
```diff
@@ -312,15 +317,15 @@ func (c cfgSub) SetContext(ctx context) {
 }
 
 func (c cfgSub) reify(opts *options) (interface{}, error) {
-	fields := c.c.fields.fields
-	arr := c.c.fields.arr
+	fields := c.c.fields.dict()
```
Comment (ruflin) [inline types.go]: `c.c.` :-)

**Pull request merged**